### ISSUE TITLE
No bindings are being generated for destructors

### ISSUE THREAD

**Issue by dflemstr** [servo/rust-bindgen#529]
This is probably mentioned elsewhere since it is such an obvious thing but I couldn't find a tracking issue for it so I thought I would create one.

### Input C/C++ Header

```C++
class Foo {
public:
    ~Foo();
};

Foo::~Foo() {
}
```

### Bindgen Invokation

```
$ bindgen test.hpp
```

### Actual Results

```rust
/* automatically generated by rust-bindgen */

#[repr(C)]
#[derive(Debug)]
pub struct Foo {
    pub _address: u8,
}
#[test]
fn bindgen_test_layout_Foo() {
    assert_eq!(::std::mem::size_of::<Foo>() , 1usize , concat ! (
               "Size of: " , stringify ! ( Foo ) ));
    assert_eq! (::std::mem::align_of::<Foo>() , 1usize , concat ! (
                "Alignment of " , stringify ! ( Foo ) ));
}
```

### Expected Results

```rust
// ... as above ...

impl Drop for Foo {
    fn drop(&mut self) {
      // ... call mangled destructor
    }
}
```

**Comment by emilio** [comment]
Yup, that's known. It shouldn't be too hard. We already keep track of the constructors.

I'd rather not implement `Drop` in bindgen itself, because we can't do it for stuff with, let's say, inlined destructors, so it can be tricky to make it work reliably, and implementing drop you prevent doing other stuff like droping the outer structure via an FFI call or similar.

**Comment by emilio** [comment]
Anyway, I'm all-in in exposing destructors. Should be easy. I'm happy to do it when I have the time, but if someone wants to jump-in and implement it, I'm happy to mentor it.

**Comment by fitzgen** [comment]
Are we done here? Can this be closed?

**Comment by emilio** [comment]
Yup! Fixed by #608
